src/commonMain/kotlin/software/momento/kotlin/sdk/responses/cache/control/CacheListResponse.kt

```diff
@@ -0,0 +1,59 @@
+package software.momento.kotlin.sdk.responses.cache.control
+
+import software.momento.kotlin.sdk.exceptions.SdkException
+
+import java.util.stream.Collectors
+
+/** Response for a list caches operation. */
+public sealed interface CacheListResponse {
+    /** A successful list caches operation. Contains the discovered caches.  */
+    public class Success(caches: List<CacheInfo>) :
```

Comment:
This can also be a data class with `caches` as a `public val`. The internal field, init block, and getter can be removed. I think IntelliJ can autogenerate an equals and hashCode to properly do a list comparison.